examples/image_classifier/mnist/torchdata/inference.py

```diff
@@ -0,0 +1,76 @@
+import asyncio
+import datetime
+
+import aiohttp
+import matplotlib.pyplot as plt
+import numpy as np
+import torch
+from torchvision import datasets, transforms
+
+# Batch_Size to kept 1 and letting torchserver to do batch inference
+BATCH_SIZE = 1
+# Torchserver model URl
+MODEL_URL = "http://127.0.0.1:8080/predictions/mnist"
+# Total number of inference calls to send in a single batch
+BATCH_TOTAL_TASKS = 100
+# Stop sending infrence request after making TOTAL_INFERENCE_CALLS
+TOTAL_INFERENCE_CALLS = BATCH_TOTAL_TASKS * 2
+
+#transform the images to tensor. Normalize the images as well.
+image_transform = transforms.Compose([
+        transforms.ToTensor(),
+        transforms.Normalize((0.1307,), (0.3081,))
+    ])
+
+# Load the dataset and set train flag to False.
+testset = datasets.MNIST('./MNIST_dataset', download=True, train=False, transform=image_transform)
+
+# Creating the dataloader.
+inference_dataset = torch.utils.data.DataLoader(testset, batch_size=BATCH_SIZE, shuffle=True)
```

Comment:
Overall I think the example looks good, it's integrating a torchvision dataset but it's not quite clearly a torchdata integration

Specifically I was hoping we could create some toy torchdata dataset directly without leveraging torchvision. I believe this change would be minor to your code but if it isn't I'm happy to merge this if you have bandwidth to work on the more vanilla torchdata integration

Reply:
I agree. @NivekT  Wondering do you have existing pipeline that they can take a reference. for vision benchmarking?

Reply:
For DataPipe reference:
1. Here is the [`torchvision` implementation of loading MNIST](https://github.com/pytorch/vision/blob/main/torchvision/prototype/datasets/_builtin/mnist.py) - it might be too complicated; one option is to use that here (similar to `datasets.MNIST`)
2. A standalone, common example is something like this:
```python
dp = FileLister(str(root), masks=[f"archive_{args.archive_size}*.tar"])
dp = dp.shuffle(buffer_size=10000)
dp = FileOpener(dp, mode="b")
dp = TarArchiveLoader(dp, mode="r:")
dp = dp.shuffle(buffer_size=archive_size)
dp = dp.sharding_filter()
dp = dp.map(pil_loader).map(pil_transformation)
# dp = dp.map(tensor_loader).map(tensor_transformation)  # Alternate - convert image to tensor then transform
```

Separately, I think we should use `DataLoader2` instead of the old version in the example. @ejguan WDYT?

Reply:
Thank you @msaroufim , @ejguan and @NivekT for your comments and guidance. I have incorporated the required changes. Looking forward to your feedback. Thank you once again.

Reply:
Thanks for the quick turnaround @PratsBhatt . Looks good. I am approving it. Minor feedback: Please link the example here since its a few levels deep and might be missed by others.  https://github.com/pytorch/serve/blob/master/examples/README.md